src/group_theory/subgroup/basic.lean

```diff
@@ -2797,17 +2799,27 @@ two subgroups of an additive group are equal."]
 def subgroup_congr (h : H = K) : H ≃* K :=
 { map_mul' :=  λ _ _, rfl, ..equiv.set_congr $ congr_arg _ h }
 
-/-- A `mul_equiv` `φ` between two groups `G` and `G'` induces a `mul_equiv` between
-a subgroup `H ≤ G` and the subgroup `φ(H) ≤ G'`. -/
-@[to_additive "An `add_equiv` `φ` between two additive groups `G` and `G'` induces an `add_equiv`
-between a subgroup `H ≤ G` and the subgroup `φ(H) ≤ G'`. "]
-def subgroup_map {G'} [group G'] (e : G ≃* G') (H : subgroup G) :
-  H ≃* H.map e.to_monoid_hom :=
-e.submonoid_map H.to_submonoid
+/-- A subgroup is isomorphic to its image under an isomorphism. If you only have an injective map,
+use `subgroup.equiv_map_of_injective`. -/
+@[to_additive  "An additive subgroup is isomorphic to its image under an an isomorphism. If you only
+have an injective map, use `add_subgroup.equiv_map_of_injective`."]
+def subgroup_map (e : G ≃* G') (H : subgroup G) : H ≃* H.map (e : G →* G') :=
+mul_equiv.submonoid_map (e : G ≃* G') H.to_submonoid
 
-end mul_equiv
+@[simp, to_additive]
+lemma coe_subgroup_map_apply (e : G ≃* G') (H : subgroup G) (g : H) :
+  ((subgroup_map e H g : H.map (e : G →* G')) : G') = e g := rfl
+
+@[simp, to_additive]
+lemma subgroup_map_symm_apply (e : G ≃* G') (H : subgroup G) (g : H.map (e : G →* G')) :
+  (e.subgroup_map H).symm g = ⟨e.symm g, set_like.mem_coe.1 $ set.mem_image_equiv.1 g.2⟩ := rfl
 
--- TODO : ↥(⊤ : subgroup H) ≃* H ?
+@[simp, to_additive]
+lemma subgroup_equiv_map_of_injective (e : G ≃* G') (H : subgroup G) :
+  H.equiv_map_of_injective (e : G →* G') (equiv_like.injective e) = subgroup_map e H :=
```

Comment:
I'm having a hard time parsing this name as a lemma and not a declaration. What do you think of

* `subgroup.equiv_map_of_injective_coe_mul_equiv`
* `subgroup.equiv_map_of_injective_mul_equiv_eq`
* `subgroup.equiv_map_of_injective_equiv_eq`
* `subgroup.equiv_map_of_injective_coe_eq`
* `mul_equiv.subgroup_equiv_map_of_injective_eq`

Reply:
I like the first one best for its unambiguity. I think it should like in the `subgroup` namespace so that the name contains `subgroup.equiv_map_of_injective`, and the `coe` atom allows separating the two other atoms `equiv_map_of_injective` and `mul_equiv`.